### PR TITLE
RONDB-756: Ported rondis to Mac OS X

### PR DESCRIPTION
**Pull request by mronstro**

### CODE REVIEW

File: pink/build.sh [hunk 1]
```diff
@@ -15,8 +15,12 @@ if [[ ! -d $SLASH_PATH ]]; then
...
+UNAME_S=`uname -s`
+if test "x$UNAME_S" = "xDarwin" ; then
+  export C_INCLUDE_PATH=/opt/homebrew_include
+  export CPLUS_INCLUDE_PATH=/opt/homebrew/include
+fi
 # Compile pink
 cd $PINK_PATH
 make SLASH_PATH=$SLASH_PATH
-cd examples && make SLASH_PATH=$SLASH_PATH
+cd rondis && make SLASH_PATH=$SLASH_PATH
```
Comment (olapiv) [inline pink/build.sh]: Technically, with the addition of Slash, this file becomes redundant now?

Comment (olapiv) [reply pink/build.sh]: No need to `cd rondis && make` - rondis is built in the parent Makefile already

File: pink/examples/Makefile [hunk 2]
```diff
@@ -28,6 +33,9 @@ SLASH_LIBRARY=$(SLASH_PATH)/slash/lib/libslash.a
 CXXFLAGS+= -I$(PINK_INCLUDE_DIR) -I$(SLASH_INCLUDE_DIR)
 
 DEP_LIBS = $(PINK_LIBRARY) $(SLASH_LIBRARY)
+ifeq ($(UNAME_S),Darwin)
+DEP_LIBS+= /Users/mikael/rondb_2410_install/lib/libprotobuf.dylib
+endif
```
Comment (olapiv) [inline pink/examples/Makefile]: RonDB is not in examples..

File: pink/rondis/Makefile [hunk 3]
```diff
@@ -28,11 +33,18 @@ ifndef RONDB_PATH
...
+RONDB_LIBRARY=$(RONDB_PATH)/lib/libndbclient.dylib
+else
 RONDB_LIBRARY=$(RONDB_PATH)/lib/libndbclient.so
+endif
 
 CXXFLAGS+= -I$(PINK_INCLUDE_DIR) -I$(SLASH_INCLUDE_DIR) -I$(RONDB_INCLUDE_DIR)
 
 DEP_LIBS = $(PINK_LIBRARY) $(SLASH_LIBRARY) $(RONDB_LIBRARY)
+ifeq ($(UNAME_S),Darwin)
+DEP_LIBS+= /opt/homebrew/lib/libprotobuf.dylib
```
Comment (olapiv) [inline pink/rondis/Makefile:46]: Wouldn't Brew dependencies be included automatically?

File: pink/rondis/rondb.cc [hunk 4]
```diff
@@ -100,6 +100,7 @@ int rondb_redis_handler(const pink::RedisCmdArgsType &argv,
                         int worker_id)
 {
     // First check non-ndb commands
+    print_args(argv);
```
Comment (olapiv) [inline pink/rondis/rondb.cc]: Can you remove this?

**Review by olapiv — CHANGES_REQUESTED** [review]

File: pink/rondis/Makefile (hunk 3 above)
Comment (mronstro) [reply pink/rondis/Makefile:46]: For some reason not, so something to figure out why that doesn't happen

File: pink/rondis/rondb.cc (hunk 4 above)
Comment (mronstro) [reply pink/rondis/rondb.cc]: ok

File: pink/build.sh (hunk 1 above)
Comment (mronstro) [reply pink/build.sh]: ok

File: pink/examples/Makefile (hunk 2 above)
Comment (mronstro) [reply pink/examples/Makefile]: ok

**Pull request closed without merging**